src/test/java/spoon/test/signature/SignatureTest.java

```diff
@@ -47,4 +51,49 @@ public void testNullSignature() throws Exception {
 		s.add(lit2);
 	}
 
+	@Test
+	public void testNullSignatureInUnboundVariable() throws Exception {
+		//Unbound variable access bug fix: 
+		//Bug description: The signature printer ignored the element Unbound variable reference 
+		//(as well all Visitor that extend CtVisitor)
+		//Fix description: modify CtVisitor (including SignaturePrinter) for visiting unbound variable access.
+		
+		
+		Factory factory = new Launcher().createFactory();
+		// We want to compile a class with an reference to a class that is not
+		// in the classpath
+		// As consequence, we set the option NoClasspath as true.
+		factory.getEnvironment().setNoClasspath(true);
+
+		String unboundVarAccess = "Complex.I";
+
+		String content = "" + "class X {" + "public Object foo() {"
+				+ " Integer.toString(" + unboundVarAccess + ");"
+				+ " return null;" + "}};";
+
+		SpoonCompiler builder = new JDTSnippetCompiler(factory, content);
+		try {
+			builder.build();
```

Comment:
Can you add a `fail()` instruction after this line? We don't would like to continue if the build is successful.